appc/appconnector.go

```diff
@@ -78,6 +80,54 @@ type RouteAdvertiser interface {
 	UnadvertiseRoute(...netip.Prefix) error
 }
 
+var (
+	metricStoreRoutesRateBuckets = []int64{1, 2, 3, 4, 5, 10, 100, 1000}
+	metricStoreRoutesNBuckets    = []int64{1, 2, 3, 4, 5, 10, 100, 1000, 10000}
+	metricStoreRoutesRate        []*clientmetric.Metric
+	metricStoreRoutesN           []*clientmetric.Metric
+)
+
+func initMetricStoreRoutes() {
+	for _, n := range metricStoreRoutesRateBuckets {
+		metricStoreRoutesRate = append(metricStoreRoutesRate, clientmetric.NewCounter(fmt.Sprintf("appc_store_routes_rate_%d", n)))
+	}
+	metricStoreRoutesRate = append(metricStoreRoutesRate, clientmetric.NewCounter("appc_store_routes_rate_over"))
+	for _, n := range metricStoreRoutesNBuckets {
+		metricStoreRoutesN = append(metricStoreRoutesN, clientmetric.NewCounter(fmt.Sprintf("appc_store_routes_n_routes_%d", n)))
+	}
+	metricStoreRoutesN = append(metricStoreRoutesN, clientmetric.NewCounter("appc_store_routes_n_routes_over"))
+}
+
+func metricStoreRoutes(rate, nRoutes int64) {
```

Comment:
skip if `rate` or `nRoutes` is 0? Maybe refactor this into one func that gets called twice with the metric slices as parameters, which would make it easier to early return if one metric is 0 but the other isn't.

Reply:
Hmm, this doesn't get called by the `rateLogger` unless the write rate is > 0. But perhaps it's getting a bit complicated? I could add a return early in that case?

Reply:
I combined the various things into a suggestion (untested, but I think it should work?), I think it comes out pretty clean.